config/templates/wso2/filter/filters.conf.njk

```diff
@@ -0,0 +1,47 @@
+# Wso2 access
+#
+[FILTER]
+    Name modify
+    Match app.{{typeTag}}.access
+    Add event.kind event
+    Add event.category web
+    Add event.dataset wso2.access{% if tmplTomcatTestAccessIndex %}
+    @metadata.index {{ tmplTomcatTestAccessIndex }}{% endif %}
+
+# Parser filter for carbon multiline core parsing
+#
+[FILTER]
+    Name parser
+    Match app.{{typeTag}}.carbon
+    Key_Name message
+    Parser parser-filter-regex-carbon-{{typeTag}}
+    Reserve_Data On
+    Preserve_Key On
+
+# App logs (carbon)
+#
+[FILTER]
+    Name modify
+    Match app.{{typeTag}}.carbon
+    Rename tomcat_time @timestamp
+    Rename tid event.provider
+    Add event.kind event
+    Add event.category web
+    Add event.dataset application.log
```

Comment:
Do we need the dataset to be carbon.log for the fingerprint?

Reply:
Matthew mentioned it using generic one, no new index with new fields needed in fingerprint. @mbystedt please correct me if I misunderstood it.

Reply:
https://www.elastic.co/guide/en/ecs/current/ecs-event.html#field-event-dataset
https://docs.wso2.com/display/ADMIN44x/Monitoring+Logs#MonitoringLogs-log_typesLogtypesinWSO2products

WSO2 refers to them as carbon. I would say it should be wso2.carbon or application.carbon.

Same index as they are generic application logs.